src/lib/symlink-manager.ts

```diff
@@ -0,0 +1,265 @@
+					if ( err instanceof Error && err.name === 'AbortError' ) {
+						return resolve();
+					}
+					reject( err );
+				}
+			} )();
+		} );
+	}
+
+	/**
+	 * Stop watching the project path for changes.
+	 */
+	async stopWatching() {
+		if ( ! this.watcherAbortController ) {
+			return;
+		}
+		this.watcherAbortController.abort( 'Watcher stopped' );
+		this.watcher = null;
+		this.watcherAbortController = null;
+	}
+
+	/**
+	 * Handle a file change event.
+	 * @param change
+	 */
+	private async handleFileChange( change: FileChangeInfo< string > ) {
+		const { filename } = change;
+
+		if ( filename === null ) {
+			return;
+		}
+
+		// Check if  the file exists
+		const fullPath = path.join( this.projectPath, filename );
+		const fileExistsOnHost = await this.fileExists( fullPath );
+
+		// If the file does not exist, it was deleted. Check if it was a symlink and remove it
+		if ( ! fileExistsOnHost && this.symlinks.has( filename ) ) {
+			await this.deleteMountedTarget( filename );
+		}
+
+		if ( ! fileExistsOnHost ) {
+			return;
+		}
+
+		const stat = await fs.lstat( fullPath );
+		if ( ! stat.isSymbolicLink() ) {
+			return;
+		}
+
+		await this.addSymlink( filename );
+	}
+
+	/**
+	 * Add a symlink to the PHP runtime by mounting the target path.
+	 * @param filename
+	 */
+	private async addSymlink( filename: string ) {
+		const fullPath = path.join( this.projectPath, filename );
+		const target = await fs.realpath( fullPath );
+
+		if ( ! this.php.requestHandler ) {
+			throw new Error( 'Request handler is not set' );
+		}
+
+		const vfsPath = path.join( this.php.requestHandler.documentRoot, filename );
+
+		// Double check to ensure the symlink exists
+		if ( ! this.php.fileExists( vfsPath ) ) {
+			return;
+		}
+
+		// Get the realpath of the symlink within the PHP runtime.
+		const vfsTarget = this.php.readlink( vfsPath );
+		this.symlinks.set( filename, vfsTarget );
+
+		await this.mountTarget( vfsTarget, target );
+	}
+
+	/**
+	 * Mount a target path in the PHP runtime.
+	 * @param vfsTarget
+	 * @param hostTarget
+	 * @private
+	 */
+	private async mountTarget( vfsTarget: string, hostTarget: string ) {
+		let mountInfo = this.mountedTargets.get( vfsTarget );
+
+		if ( mountInfo === undefined ) {
+			mountInfo = {
+				preExisting: this.php.fileExists( vfsTarget ),
+				referenceCount: 0,
+				unmountFunction: null,
+			};
+			this.mountedTargets.set( vfsTarget, mountInfo );
+		}
+
+		mountInfo.referenceCount++;
+
+		if ( mountInfo.preExisting || mountInfo.unmountFunction !== null ) {
+			return;
+		}
+
+		this.php.mkdir( vfsTarget );
+		mountInfo.unmountFunction = await this.php.mount(
+			vfsTarget,
+			createNodeFsMountHandler( hostTarget )
+		);
+	}
+
+	/**
+	 * Delete a symlink from the PHP runtime.
+	 *
+	 * @param filename
+	 */
+	private async deleteMountedTarget( filename: string ) {
+		const vfsTarget = this.symlinks.get( filename );
+
+		if ( vfsTarget === undefined ) {
```

Comment:
In other areas of the code, we check undefined values with the following approach:

```suggestion
		if ( typeof vfsTarget === 'undefined' ) {
```